x/migration/keeper/msg_server_claim_morse_acount_test.go

```diff
@@ -0,0 +1,197 @@
+	}
+}
+
+func TestMsgServer_ClaimMorseAccount_Error(t *testing.T) {
+	k, ctx := keepertest.MigrationKeeper(t)
+	srv := keeper.NewMsgServerImpl(k)
+
+	// Generate and import a Morse claimable account.
+	numAccounts := 1
+	_, accountState := testmigration.NewMorseStateExportAndAccountState(t, numAccounts)
+	accountStateHash, err := accountState.GetHash()
+	require.NoError(t, err)
+
+	_, err = srv.ImportMorseClaimableAccounts(ctx, &migrationtypes.MsgImportMorseClaimableAccounts{
+		Authority:             authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		MorseAccountState:     *accountState,
+		MorseAccountStateHash: accountStateHash,
+	})
+	require.NoError(t, err)
+
+	// Generate the corresponding morse private key using the account slice index as a seed.
+	morsePrivKey := testmigration.NewMorsePrivateKey(t, 0)
+
+	// Claim the MorseClaimableAccount with a random Shannon address.
+	msgClaim, err := migrationtypes.NewMsgClaimMorseAccount(
+		sample.AccAddress(),
+		accountState.Accounts[0].GetMorseSrcAddress(),
+		morsePrivKey,
+	)
+	require.NoError(t, err)
+
+	t.Run("invalid claim msg", func(t *testing.T) {
+		// Copy the message and set the morse signature to an empty string.
+		invalidMsgClaim := *msgClaim
+		invalidMsgClaim.MorseSignature = nil
+
+		expectedErr := status.Error(
+			codes.InvalidArgument,
+			migrationtypes.ErrMorseAccountClaim.Wrapf(
+				"invalid morseSignature length (0): %q", "",
+			).Error(),
+		)
+
+		_, err := srv.ClaimMorseAccount(ctx, &invalidMsgClaim)
+		require.EqualError(t, err, expectedErr.Error())
+	})
+
+	t.Run("account not found", func(t *testing.T) {
+		// Copy the message and set the morse src address to a valid but incorrect address.
+		invalidMsgClaim := *msgClaim
+		invalidMsgClaim.MorseSrcAddress = sample.MorseAddressHex()
+
+		expectedErr := status.Error(
+			codes.NotFound,
+			migrationtypes.ErrMorseAccountClaim.Wrapf(
+				"no morse claimable account exists with address %q",
+				invalidMsgClaim.GetMorseSrcAddress(),
+			).Error(),
+		)
+
+		_, err := srv.ClaimMorseAccount(ctx, &invalidMsgClaim)
+		require.EqualError(t, err, expectedErr.Error())
+	})
+
+	t.Run("account already claimed (non-zero claimed_at_height)", func(t *testing.T) {
+		// Set the claimed at height BUT NOT the Shannon destination address.
+		morseClaimableAccount := *accountState.Accounts[0]
+		morseClaimableAccount.ClaimedAtHeight = 10
+		k.SetMorseClaimableAccount(ctx, morseClaimableAccount)
+
+		expectedErr := status.Error(
+			codes.FailedPrecondition,
+			migrationtypes.ErrMorseAccountClaim.Wrapf(
+				"morse address %q has already been claimed at height %d by shannon address %q",
+				accountState.Accounts[0].GetMorseSrcAddress(),
+				10,
+				accountState.Accounts[0].GetShannonDestAddress(),
+			).Error(),
+		)
+
+		_, err := srv.ClaimMorseAccount(ctx, msgClaim)
+		require.EqualError(t, err, expectedErr.Error())
+	})
+
+	t.Run("account already claimed (non-empty shannon_dest_address)", func(t *testing.T) {
```

Comment:
Please add tests for:

1. Failure if supplier stake is non-zero
2. Failure if application stake is non-zero